File: library/src/main/java/com/nasahapps/nasahutils/app/LoaderResult.java
package com.nasahapps.nasahutils.app;

/**
 * Created by Hakeem on 2/14/16.
 */
public class LoaderResult<D> {

    public D data;
    public Throwable error;

    public LoaderResult() {
    }

    public LoaderResult(Throwable error) {
        this.error = error;
    }

    public LoaderResult(D data) {
        this.data = data;
    }
}

File: library/src/main/java/com/nasahapps/nasahutils/app/LoaderResult.kt
package com.nasahapps.nasahutils.app

/**
 * Created by Hakeem on 2/14/16.
 */
class LoaderResult<D> {

    var data: D? = null
    var error: Throwable? = null

    constructor()

    constructor(error: Throwable) {
        this.error = error
    }

    constructor(data: D) {
        this.data = data
    }

}
